SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -75,26 +103,24 @@ void USpatialLoadBalanceEnforcer::ProcessQueuedAclAssignmentRequests()
 
 	for (WriteAuthAssignmentRequest& Request : AclWriteAuthAssignmentRequests)
 	{
-		static const int32 WarnOnAttemptNum = 5;
-		if (Request.ProcessAttempts >= WarnOnAttemptNum)
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Failed to process WriteAuthAssignmentRequest with EntityID: %lld. Process attempts made: %d"), Request.EntityId, Request.ProcessAttempts);
-		}
-
-		Request.ProcessAttempts++;
-
-		// TODO - if some entities won't have the component we should detect that before queueing the request.
-		// Need to be certain it is invalid to get here before receiving the AuthIntentComponent for an entity, then we can check() on it.
-		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
+ 		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
 		if (AuthorityIntentComponent == nullptr)
 		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Detected entity without AuthIntent component. EntityId: %lld"), Request.EntityId);
-			continue;
+			// TODO(zoning): This is possible if the ACL update lags behind the intent update. Evaluate whether this is a serious problem.
+			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("(%s) Entity without AuthIntent component will not be processed. EntityId: %lld"), *WorkerId, Request.EntityId);
+			CompletedRequests.Add(Request.EntityId);
+			return;
 		}
 
 		const FString* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
 		if (OwningWorkerId == nullptr)
 		{
+			const int32 WarnOnAttemptNum = 5;
+			Request.ProcessAttempts++;
+			if (Request.ProcessAttempts >= WarnOnAttemptNum)
+			{
+				UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Failed to process WriteAuthAssignmentRequest because virtual worker mapping is unset for EntityID: %lld. Process attempts made: %d"), Request.EntityId, Request.ProcessAttempts);
```

Comment:
Should we reset `Request.ProcessAttempts` to 0 here? Otherwise after 5 attempts it will print Warning for every new failed attempt.

Reply:
Might be useful to know the exact number of attempts made. We could do a modulo to reduce the log spam? I was thinking about adding the mod but didn't want to for the extra cost of mod but maybe I was being a bit silly.

Reply:
I like the idea to use mod 👍

Reply:
mod added.